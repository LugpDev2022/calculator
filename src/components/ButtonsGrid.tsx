import { Dispatch } from 'react';
import { CalcAction } from '../calculatorReducer';
import Button from './Button';

interface Props {
  isThemeDark: boolean;
  dispatch: Dispatch<CalcAction>;
  operation: string;
}

const buttonData = [
  ['(', ')', 'DEL', 'AC'],
  ['7', '8', '9', 'รท'],
  ['4', '5', '6', 'ร'],
  ['1', '2', '3', '+'],
  ['0', '.', '=', '-'],
];

const ButtonsGrid: React.FC<Props> = ({ isThemeDark, dispatch, operation }) => {
  return (
    <div className='grid grid-cols-4 gap-2'>
      {buttonData.map((row) =>
        row.map((buttonText) => (
          <Button
            key={buttonText}
            isThemeDark={isThemeDark}
            operation={operation}
            dispatch={dispatch}
          >
            {buttonText}
          </Button>
        ))
      )}
    </div>
  );
};

export default ButtonsGrid;
