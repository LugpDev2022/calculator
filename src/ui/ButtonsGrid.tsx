import { Button, ResultButton } from '../components';
import { symbols } from '../assets/symbols';

export const ButtonsGrid = () => (
  <div className='row mt-1'>
    <div className='col-12'>
      <div className='d-flex flex-wrap justify-content-between'>
        {symbols.map(symbol => {
          return <Button key={symbol} buttonSymbol={symbol} />;
        })}
        <ResultButton />
      </div>
    </div>
  </div>
);
