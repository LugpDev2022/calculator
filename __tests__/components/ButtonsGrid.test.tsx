import React from 'react';
import { vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import ButtonsGrid from '../../src/components/ButtonsGrid';

vi.mock('../../src/components/Button', () => ({
  default: ({ keySign }) => <button>{keySign}</button>,
}));

describe('Tests on <ButtonsGrid />', () => {
  test('should match snapshot', () => {
    const { container } = render(<ButtonsGrid />);

    expect(container).toMatchSnapshot();
  });

  test('should render 20 buttons', () => {
    render(<ButtonsGrid />);

    const buttons = screen.getAllByRole('button');
    expect(buttons.length).toBe(20);
  });
});
