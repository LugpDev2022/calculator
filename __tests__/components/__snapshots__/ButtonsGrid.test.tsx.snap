// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests on <ButtonsGrid /> > should match snapshot 1`] = `
<div>
  <div
    class="grid grid-cols-4 gap-2"
  >
    <button>
      (
    </button>
    <button>
      )
    </button>
    <button>
      DEL
    </button>
    <button>
      AC
    </button>
    <button>
      7
    </button>
    <button>
      8
    </button>
    <button>
      9
    </button>
    <button>
      รท
    </button>
    <button>
      4
    </button>
    <button>
      5
    </button>
    <button>
      6
    </button>
    <button>
      ร
    </button>
    <button>
      1
    </button>
    <button>
      2
    </button>
    <button>
      3
    </button>
    <button>
      +
    </button>
    <button>
      0
    </button>
    <button>
      .
    </button>
    <button>
      =
    </button>
    <button>
      -
    </button>
  </div>
</div>
`;
